File: src/Functions.Worker.Extensions.OpenAI/Assistants/AssistantPostRequest.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Microsoft.Azure.Functions.Worker.Extensions.OpenAI.Assistants;

/// <summary>
/// Assistant post request which is used to relay post requests.
/// </summary>
public class AssistantPostRequest
{
    /// <summary>
    /// Gets user message that user has entered for assistant to respond to.
    /// </summary>
    public string? UserMessage { get; set; }

    /// <summary>
    /// Gets the ID of the assistant to update.
    /// </summary>
    public string Id { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the OpenAI chat model to use.
    /// </summary>
    /// <remarks>
    /// When using Azure OpenAI, then should be the name of the model <em>deployment</em>.
    /// </remarks>
    public string? Model { get; set; }
}

File: java-library/src/main/java/com/microsoft/azure/functions/openai/annotation/AssistantPostRequest.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package com.microsoft.azure.functions.openai.annotation;

/**
 * Assistant post request which is used to relay post requests.
 */
public class AssistantPostRequest {

    private String userMessage;
    private String id = "";
    private String model;

    /**
     * Initializes a new instance of the AssistantPostRequest class.
     */
    public AssistantPostRequest(String userMessage, String id, String model) {
        this.userMessage = userMessage;
        this.id = id;
        this.model = model;
    }

    /**
     * Initializes a new instance of the AssistantPostRequest class.
     */
    public AssistantPostRequest(String userMessage, String id) {
        this.userMessage = userMessage;
        this.id = id;
    }

    /**
     * Gets user message that user has entered for assistant to respond to.
     */
    public String getUserMessage() {
        return userMessage;
    }

    /**
     * Sets user message that user has entered for assistant to respond to.
     */
    public void setUserMessage(String userMessage) {
        this.userMessage = userMessage;
    }

    /**
     * Gets the ID of the assistant to update.
     */
    public String getId() {
        return id;
    }

    /**
     * Sets the ID of the assistant to update.
     */
    public void setId(String id) {
        this.id = id;
    }

    /**
     * Gets the OpenAI chat model to use.
     * <p>
     * When using Azure OpenAI, then should be the name of the model <em>deployment</em>.
     * </p>
     */
    public String getModel() {
        return model;
    }

    /**
     * Sets the OpenAI chat model to use.
     * <p>
     * When using Azure OpenAI, then should be the name of the model <em>deployment</em>.
     * </p>
     */
    public void setModel(String model) {
        this.model = model;
    }
}
